### PR TITLE
implementations: add a list of known implementations

### LINKED ISSUE #126: [1.0.0] Need actual independent implementations
Currently we don't have any canonical independent implementations of the specification. Docker's image format is different to ours (even though the base came from them), and rkt doesn't have support for ours.

As a result, this spec is entirely untested as we don't have any code implementing the spec word-for-word. As [several people have discussed on the mailing list](https://groups.google.com/a/opencontainers.org/forum/#!topic/dev/NPBMTrniG8Q), this should be considered a blocking issue for 1.0.

Personally, I would want to have two implementations of the image-spec (like we do with the runtime-spec) before we can consider this ready. Preferably these implementations would be part of Docker and rkt (and have a reasonable cook time).

### LINKED ISSUE #396: Build story: producing an OCI image
The image spec is really awesome and I can't wait to see a stable release of the spec, integrated everywhere.

What I'm personally missing right now is a way to produce (build, compose, orchestrate; name it the way you want) an image: I mean a spec for steps needed to create an image. I'm sure that's very much out of scope of this project. What I'm interested in is
1. if such spec will ever be part of OCI
2. or if there are any plans in this area
3. or if this topic is really up to vendors to deal with

I really would like to know @vbatts' ideas about this.

### PR DESCRIPTION
**Pull request by vbatts**
Fixes #126
Fixes #396 

Signed-off-by: Vincent Batts <vbatts@hashbangbash.com>

### CODE REVIEW

File: implementations.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (jonboulle) [inline implementations.md]: is this newline necessary?

File: implementations.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+* [cloudfoundry/grootfs](https://github.com/cloudfoundry/grootfs) ([source](https://github.com/cloudfoundry/grootfs/blob/c3da26e1e463b51be1add289032f3dca6698b335/fetcher/remote/docker_src.go))
+* [Mesos plans](https://issues.apache.org/jira/browse/MESOS-5011) ([design doc](https://docs.google.com/document/d/1Pus7D-inIBoLSIPyu3rl_apxvUhtp3rp0_b0Ttr2Xww/edit#heading=h.hrvk2wboog4p))
+* [Docker](https://github.com/docker)
+ - [`docker save/load` WIP](https://github.com/docker/docker/pull/26369)
+ - [docker registry PR](https://github.com/docker/distribution/pull/2076)
+* [Containers](https://github.com/containers/)
+ - [containers/build](https://github.com/containers/build)
+ - [containers/image](https://github.com/containers/image)
+* [coreos/rkt](https://github.com/coreos/rkt)
+* [erikh/box](https://github.com/erikh/box)
```
Comment (jonboulle) [inline implementations.md]: Maybe add a suggestion for people to add their implementations?

**Comment by jonboulle** [comment]
thanks for doing this

File: implementations.md (hunk 2 above)
Comment (vbatts) [inline implementations.md]: yea, fair

File: implementations.md (hunk 1 above)
Comment (vbatts) [inline implementations.md]: noup

**Comment by vbatts** [comment]
updated. PTAL

**Comment by jonboulle** [comment]
seems reasonable

**Comment by jonboulle** [comment]
LGTM, obviously the idea this is a living document, so we can keep an eye on keeping it up to date as the ecosystem evolves.

File: implementations.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# OCI Image Implementations
+
+Projects or Companies currently adopting the OCI Image Specification
+
+* [projectatomic/skopeo](https://github.com/projectatomic/skopeo)
+* [Amazon Elastic Container Registry (ECR)](https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-manifest-formats.html) ([announcement](https://aws.amazon.com/about-aws/whats-new/2017/01/amazon-ecr-supports-docker-image-manifest-v2-schema-2/))
+* [openSUSE/umoci](https://github.com/openSUSE/umoci)
+* [cloudfoundry/grootfs](https://github.com/cloudfoundry/grootfs) ([source](https://github.com/cloudfoundry/grootfs/blob/c3da26e1e463b51be1add289032f3dca6698b335/fetcher/remote/docker_src.go))
+* [Mesos plans](https://issues.apache.org/jira/browse/MESOS-5011) ([design doc](https://docs.google.com/document/d/1Pus7D-inIBoLSIPyu3rl_apxvUhtp3rp0_b0Ttr2Xww/edit#heading=h.hrvk2wboog4p))
+* [Docker](https://github.com/docker)
```
Comment (stevvooe) [inline implementations.md]: Don't forget containerd!

Comment (vbatts) [inline implementations.md]: oh right. I knew the that image-spec support was on the horizon, but didn't think it was a PR yet. Should've looked

Comment (vbatts) [inline implementations.md]: updated

**Comment by stevvooe** [comment]
LGTM

**Comment by jonboulle** [comment]
lgtm

**Pull request merged**